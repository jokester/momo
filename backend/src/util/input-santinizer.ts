import { Either, left, right } from 'fp-ts/lib/Either';

function sanitizeEmail(email: string): Either<string, string> {
  const trimmed = (email || '').trim().toLowerCase();

  if (/[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,6}/i.test(trimmed)) {
    return right(trimmed);
  }
  return left('email malformed');
}

function sanitizePass(password: string): Either<string, string> {
  if (password && password.length > 6) {
    return right(password);
  }
  return left('password too short');
}

function userId(orig: unknown) {
  if (typeof orig === 'string' && / /.test(orig)) {
    return right(orig);
  }
  return left('incorrect userId');
}

export const Sanitize = {
  email: sanitizeEmail,
  pass: sanitizePass,
  userId,

  isString(x: unknown): x is string {
    return typeof x === 'string';
  },
} as const;
