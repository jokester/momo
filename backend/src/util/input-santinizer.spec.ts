import { Sanitize } from "./input-santinizer";

describe('Sanitize', () => {
  it('launders email', () => {
      expect(Sanitize.email('momocraft@gmail.com')).toMatchSnapshot()
    }
  )
})
