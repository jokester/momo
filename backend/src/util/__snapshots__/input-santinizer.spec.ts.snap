// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sanitize launders email 1`] = `
Object {
  "_tag": "Right",
  "right": "momocraft@gmail.com",
}
`;
